src/test/java/com/fasterxml/jackson/dataformat/xml/ser/XmlPrettyPrinterTest.java

```diff
@@ -188,4 +189,76 @@ public void testMultiLevel172() throws Exception
                 +"</Company>\n",
                 xml);
     }
+
+    public void testLineFeed_withCustomLineFeed() throws Exception {
+        Company root = new Company();
+        root.employee.add(new Employee("abc"));
+
+        String xml = _xmlMapper.writer()
+            .with(new DefaultXmlPrettyPrinter().withCustomLineFeed("\n\rLF\n\r"))
+            .with(ToXmlGenerator.Feature.WRITE_XML_DECLARATION)
+            .writeValueAsString(root);
+        // unify possible apostrophes to quotes
+        xml = a2q(xml);
+
+        // with indentation, should get linefeeds in prolog/epilog too
+        assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                + "<Company>\n\rLF\n\r"
+                + "  <e>\n\rLF\n\r"
+                + "    <employee>\n\rLF\n\r"
+                + "      <id>abc</id>\n\rLF\n\r"
+                + "      <type>FULL_TIME</type>\n\rLF\n\r"
+                + "    </employee>\n\rLF\n\r"
+                + "  </e>\n\rLF\n\r"
+                + "</Company>\n",
```

Comment:
I am not sure if this line should also end with customLineFeed also?

Reply:
yes - all the line feeds should be the same

actually, could we rename the method as `withCustomNewLine`? - I don't think 'line feed' is a good name for this property at all - 'new line' is better

Reply:
> yes - all the line feeds should be the same

noted will make sure tests are applied there?

> actually, could we rename the method as withCustomNewLine? - I don't think 'line feed' is a good name for this property at all - 'new line' is better

Agreed and I will change names as below!🙏🏼
- `_lineFeed` -> `_newLine`
- `DEFAULT_LINE_FEED` -> `DEFAULT_NEW_LINE`
- `withCustomLineFeed` -> `withCustomNewLine`

Reply:
@pjfanning So here is an update.

> actually, could we rename the method as withCustomNewLine? - I don't think 'line feed' is a good name for this property at all - 'new line' is better

Done.
All names are changed from `line feed` to `new line`

> yes - all the line feeds should be the same

 May I get your opinion here? It seems like writing custom `new lines` before and after root element might be out of this PR's scope. The 'new lines' before and after the root element are written by method `writePrologLinefeed()` of another instance of `DefaultXmlPrettyPrinter`.

 This instance invoking `writePrologLinefeed()` instance seems to be initialized and used inside ToXmlGenerator which we cannot hook into from neither `XmlMapper().setDefaultPrettyPrinter()` nor `ObjectWriter.with(PrettyPrinter)`.

For references,
- Similar issues I found doing research. #554 
-  Check 👉🏼 [the code here writes BEFORE root element](https://github.com/FasterXML/jackson-dataformat-xml/blob/0e59be67bd3e9cfcd73b2b62a95bcc0f5b2dda3c/src/main/java/com/fasterxml/jackson/dataformat/xml/ser/ToXmlGenerator.java#L258) and [the code here writes AFTER root element](https://github.com/FasterXML/jackson-dataformat-xml/blob/0e59be67bd3e9cfcd73b2b62a95bcc0f5b2dda3c/src/main/java/com/fasterxml/jackson/dataformat/xml/ser/ToXmlGenerator.java#L619)

Reply:
It makes little sense to me not to use the same new lines throughout the XML doc. I'm now -1 on this PR.

Reply:
> It makes little sense to me not to use the same new lines throughout the XML doc. I'm now -1 on this PR.

Agreed and -1 from me, too. 
Having a different new line configuration it makes little sense to me also.
I will try to check if a reasonable solution exists before actually closing this PR.
Thank you for your review, always! 🙏🏼🙏🏼

Reply:
Is the discrepancy between prolog/epilog and regular content due to Stax API not exposing write methods?
If I remember correctly there was something odd about that handling; maybe I can help here.

Reply:
Oh no actually looks like there is `XmlPrettyPrinter` method `writePrologLinefeed()` which handles this. As well as possible related `writeRootValueSeparator()`.

Reply:
> Oh no actually looks like there is `XmlPrettyPrinter` method `writePrologLinefeed()` which handles this. As well as possible related `writeRootValueSeparator()`.

@cowtowncoder The BIG (I think) problem here is that the `DefaultXmlPrettyPrinter` instance running `writePrologLinefeed()` is not same as the one set by `_xmlMapper..with(new DefaultXmlPrettyPrinter().withCustomNewLine("\n\rLF\n\r"))`

Reply:
@JooHyukKim Ok that sounds like a problem indeed! I wonder if it's the case of `Indenter` instance not being created yet (with `createInstance()`) or what is happening.

Reply:
@cowtowncoder 
Not the indenter, actually. There is a separate instance of `DefaultXmlPrettyPrinter` initialized together with `ToXmlGenerator` that invokes `_xmlPrettyPrinter.writePrologLinefeed` to write prolog and epilog.

You can refer to below to code lines.
- [ToXmlGenerator. initGenerator](https://github.com/FasterXML/jackson-dataformat-xml/blob/0e59be67bd3e9cfcd73b2b62a95bcc0f5b2dda3c/src/main/java/com/fasterxml/jackson/dataformat/xml/ser/ToXmlGenerator.java#L258) method for prolog and
- [ToXmlGenerator._handleObject](https://github.com/FasterXML/jackson-dataformat-xml/blob/0e59be67bd3e9cfcd73b2b62a95bcc0f5b2dda3c/src/main/java/com/fasterxml/jackson/dataformat/xml/ser/ToXmlGenerator.java#L619) for epilog

My first (probable) observation is that writing prolog and epilog with instance of `DefaultXmlPrettyPrinter` I mentioned above seems NOT to be overridden by `ObjectMapper.setDefaultPrettyPrinter(PrettyPrinter)` nor `ObjectWriter.with(PrettyPrinter)`.

Second observation is `ToXmlGenerator` writes prolog during the process of initialization within `ToXmlGenerator.initGenerator` method.

It will be great, @cowtowncoder if you can help out a little here? To override DefaultXmlPrettyPrinter of `ToXmlGenerator._xmlPrettyPrinter` before it writes prolog and epilog?

Reply:
Right, pretty printer instance. I will try to see if I can figure it out -- the problem is that XML module extends basic `PrettyPrinter` API to its needs but databind still tries to orchestrate handling. So I will try to see if I can figure out how to avoid there being different instance being used.

Reply:
@JooHyukKim Ah. Part of the problem is that you haven't updated copy constructor:

    protected DefaultXmlPrettyPrinter(DefaultXmlPrettyPrinter base) {  }

to copy `_newLine` property -- so when method 

```
    @Override
    public DefaultXmlPrettyPrinter createInstance() {
```

is called the new instance "forgets" `_newLine`. So you need to add 

```
    _newLine = base._newLine;
```

in that constructor.

Reply:
> is called the new instance "forgets" `_newLine`. So you need to add

Woooooow thank you for helping out 🥹 @cowtowncoder 
How did I miss that!
Done 🙏🏼